alcs/acs2/AbstractCondition.py

```diff
@@ -0,0 +1,40 @@
+from alcs import Perception
+from alcs.acs2 import default_configuration
+
+
+class AbstractCondition(list):
+    def __init__(self, seq=(), cfg=None):
+        self.cfg = cfg or default_configuration
+        if not seq:
+            list.__init__(
+                self,
+                [self.cfg.classifier_wildcard] * self.cfg.classifier_length)
+        else:
+            list.__init__(self, seq)
+            if len(self) != self.cfg.classifier_length:
+                raise ValueError('Illegal length of effect string')
```

Comment:
In the abstracted version it shouldn't be called `effect`. Please use *perception* instead. Also a few lines below.

Reply:
Agree.